src/models/Settings.js

```diff
@@ -275,7 +277,16 @@ function (Okta, Q, Errors, BrowserFeatures, Util, Logger, OAuth2Util, config) {
               },
               click: function (e) {
                 e.preventDefault();
-                OAuth2Util.getTokens(self, {idp: idp.id});
+                if (self.get('oauth2Enabled')) {
```

Comment:
I can't remember exactly the order of how derived props are computed, but I think we should add `oauth2Enabled` as a dependency in line 263 `deps: ['configuredSocialIdps', 'customButtons', 'oauth2Enabled'],`, so we know that it is computed before `configuredButtons`

Reply:
Good catch! Thanks!

Reply:
Maybe... but in `OAuth2Util` you don't really get any derived property I think, it is either regular props or the `authClient`. 
Since I don't know exactly how we compute derived props, I think it is safer to move it to deps. I assume that if a derived prop depends on another derived prop, the ones that have no dependency will be computed first.

Reply:
- actually derived props are computed when construct the model.
- the diff with regular props are derived props adapts to changes from regular props. (update derived props when deps changed)
- now the question is if `oauth2Enabled` has been changed, shall we update `configuredButtons` and will it impact the click handler? yes for 1 but properly no for 2 since once the `configuredButtons` have been rendered, we didn't re-render upon changes.
- theoretically it would be more accurate to always the derived prop at click handler cause it could always get latest value v.s. adding the `oauth2Enabled` as another dependency which would become a closure and won't be updated after first render.
- I'm thinking is we shall't create event handlers at Model but View and maybe shall refactor the code at some point?

Reply:
> theoretically it would be more accurate to always the derived prop at click handler cause it could always get latest value

good point. I think then this is good... as you said, we probably shouldn't have click handlers here. But I agree, that is a different task than we can deal with later.